plugins/metrics/message.go

```diff
@@ -1,131 +1,98 @@
-
-// measured value of the received MPS
-var measuredReceivedMPS uint64
+var (
+	// Total number of processed messages since start of the node.
+	messageTotalCount atomic.Uint64
 
-// Total number of processed messages since start of the node.
-var messageTotalCount uint64
+	// current number of message tips.
+	messageTips atomic.Uint64
 
-// current number of message tips
-var messageTips uint64
+	// counter for the received MPS
+	mpsReceivedSinceLastMeasurement atomic.Uint64
 
-// MPS figures for different payload types
-var mpsPerPayloadSinceLastMeasurement = make(map[payload.Type]uint64)
-var mpsPerPayloadMeasured = make(map[payload.Type]uint64)
+	// measured value of the received MPS
+	measuredReceivedMPS atomic.Uint64
 
-// Number of messages per payload type since start of the node.
-var messageCountPerPayload = make(map[payload.Type]uint64)
+	// Number of messages per payload type since start of the node.
+	messageCountPerPayload = make(map[payload.Type]uint64)
 
-// protect maps from concurrent read/write
-var messageLock syncutils.RWMutex
+	// protect map from concurrent read/write.
+	messageCountPerPayloadMutex syncutils.RWMutex
+)
 
 ////// Exported functions to obtain metrics from outside //////
 
 // MessageTotalCount returns the total number of messages seen since the start of the node.
 func MessageTotalCount() uint64 {
-	return atomic.LoadUint64(&messageTotalCount)
-}
-
-// MPS retrieves the current messages per second number.
-func MPS() uint64 {
-	return atomic.LoadUint64(&measuredReceivedMPS)
+	return messageTotalCount.Load()
 }
 
 // MessageCountPerPayload returns a map of message payload types and their count since the start of the node.
 func MessageCountPerPayload() map[payload.Type]uint64 {
-	messageLock.RLock()
-	defer messageLock.RUnlock()
+	messageCountPerPayloadMutex.RLock()
+	defer messageCountPerPayloadMutex.RUnlock()
 
 	// copy the original map
-	target := make(map[payload.Type]uint64)
+	copy := make(map[payload.Type]uint64)
 	for key, element := range messageCountPerPayload {
-		target[key] = element
+		copy[key] = element
 	}
 
-	return target
+	return copy
 }
 
-// MPSPerPayload returns a map of message payload types and their corresponding MPS values.
-func MPSPerPayload() map[payload.Type]uint64 {
-	messageLock.RLock()
-	defer messageLock.RUnlock()
+// MessageTips returns the actual number of tips in the message tangle.
+func MessageTips() uint64 {
+	return messageTips.Load()
+}
 
-	// copy the original map
-	target := make(map[payload.Type]uint64)
-	for key, element := range mpsPerPayloadMeasured {
-		target[key] = element
-	}
+////// Handling data updates and measuring //////
 
-	return target
+func increasePerPayloadCounter(p payload.Type) {
+	messageCountPerPayloadMutex.Lock()
+	defer messageCountPerPayloadMutex.Unlock()
+	// init counter with zero value if we see the payload for the first time
+	if _, exist := messageCountPerPayload[p]; !exist {
+		messageCountPerPayload[p] = 0
+	}
```

Comment:
This is not needed since when you assign `messageCountPerPayload[p]++` Go will operate on the default zero value for the given type.